superset-frontend/src/cccs-viz/plugins/plugin-chart-ag-grid/src/ag-grid/ContextMenu/MenuItems/DownloadEmailMenuItem.tsx

```diff
@@ -28,36 +29,43 @@ export default function DownloadEmailMenuItem(
   const dispatch = useDispatch();
 
   const onClick = () => {
-    const endpoint = `/api/v1/fission/get-eml?file=${props.data}`;
-    dispatch(addInfoToast('Download started'));
+    props.data.forEach(fileName => {
+      const endpoint = `/api/v1/fission/get-eml?file=${fileName}`;
+      dispatch(addInfoToast('Download started'));
 
-    SupersetClient.get({ endpoint, timeout: QUERY_TIMEOUT_LIMIT })
-      .then(({ json }) => {
-        if (json.result?.content?.indexOf('base64') !== -1) {
-          // Check for base64 encoding
-          const b64Data = json.result.content.split(',')[1];
-          const contentType = 'message/rfc822'; // Correct MIME type for EML files
+      SupersetClient.get({ endpoint, timeout: QUERY_TIMEOUT_LIMIT })
+        .then(({ json }) => {
+          if (json.result?.content?.indexOf('base64') !== -1) {
+            // Check for base64 encoding
+            const b64Data = json.result.content.split(',')[1];
+            const contentType = 'message/rfc822'; // Correct MIME type for EML files
 
-          // Convert base64 data to a blob with the appropriate MIME type
-          const blob = b64ToBlob(b64Data, contentType);
+            // Convert base64 data to a blob with the appropriate MIME type
+            const blob = b64ToBlob(b64Data, contentType);
 
-          // Derive file name from the title or use a default name, ensuring it ends with .eml
-          const uniqueTitle = json.result.title
-            ? `${json.result.title}.eml`
-            : `${props.data}.eml`;
+            // Derive file name from the title or use a default name, ensuring it ends with .eml
+            const uniqueTitle = json.result.title
+              ? `${json.result.title}.eml`
+              : `${fileName}.eml`;
 
-          // Use FileSaver or a similar library to trigger the file download
-          saveAs(blob, uniqueTitle);
-        } else if (json.result?.content) {
-          dispatch(addDangerToast('Invalid file format.'));
-        } else {
-          dispatch(addDangerToast('No content to download.'));
-        }
-      })
-      .catch(error => {
-        const errorMessage = `Download failed: ${error.message}`;
-        dispatch(addDangerToast(errorMessage));
-      });
+            // Use FileSaver or a similar library to trigger the file download
+            saveAs(blob, uniqueTitle);
+          } else if (json.result?.content) {
+            dispatch(addDangerToast('Invalid file format.'));
+          } else {
+            dispatch(addDangerToast('No content to download.'));
+          }
+        })
+        .catch(error => {
+          let errorMessage;
+          if (error.content === 'undefined') {
```

Comment:
Can we use a HTTP status code rather than the error text to check if the file is not found, assuming the endpoint returns 404 if the file is not found.